### DISCUSSION TITLE
Adding body battery and respiration rate

### DISCUSSION THREAD

**Discussion by lpierron** [tcgoetz/GarminDB#134] [Ideas]
It will be great to add body battery and respiration rate information.
But I ask myself how to add these features, without broken the database.
Any idea ?

**Comment by tcgoetz** [comment]
Respiration rate is already present in the DB. Adding body battery would be a incompatible change and require the DBs to be recreated.

**Comment by lpierron** [comment]
Thanks. I have to look about the respiration rate. 
Maybe, I can check how to add body battery when I will recreate the DB.
I want to recreate the DB, because missing one month, but I postpone because I added a lot of views to extract what I think is pertinent for me and don't know how to export all my views and reimport it easily.

You did a great job with GarminDB, thanks a lot !

**Reply by tcgoetz** [reply]
I've been planning on adding a plugin interface to the JSON file processing (right now plugins are just for FIT files). If I do that you could use that to process body battery into an additional table. An additional table would not be an incompatible change.
